### DISCUSSION TITLE
Issues with clients behind a WebProxy

### DISCUSSION THREAD

**Discussion by devNan0** [amidaware/tacticalrmm#1695] [Q&A]
Hey there,

I'm currently testing RMM and discovered that Agents behind a WebProxy aren't working correctly.
Installation works fine, but then the Agent ist marked as Offline in the WebUI.

Via the MeshCentral i can see, take control over and show the local files of the Client, but in the RMM WebUI it show him as Offline with no Hardware Info, Agent v0.1.0 and last boot time 54 years ago.

I tried running the Agent with logging enabled but there are no errors only HTTP 200 logging.
Also tried disabling Antivirus, which didn't help. Clients without the WebProxy are working fine.

Does anyone here has some hints on what to do for further debugging?

Thanks in Advance!

**Comment by dinger1986** [comment]
do you mean you have an additional proxy added? Or those agents use a proxy to access the internet?

**Reply by devNan0** [reply]
Those agents use a proxy to access the internet.
I also tried installing the agent with the `--proxy` setting.

The RMM server doesn't have a proxy in front of it.

**Reply by silversword411** [reply]
Your proxy is probably filtering the data, or not supporting the packet payloads.
Your proxy has to forward TLS traffic without MITM modification, and also support websockets. And not prematurely slice session times.

What outbound web proxy are you pushing your RMM agents traffic thru?

**Reply by wh1te909** [reply]
The `--proxy` implementation is broken atm doesn't work with NATS I need to fix that

**Reply by devNan0** [reply]
@wh1te909 thanks for the Info, would an older agent work? I just won't to test.

**Reply by wh1te909** [reply]
no, an older agent will not work. nats client currently does not have built in support for http proxy so we will need implement a custom dialer. in the meantime the workaround is to use the old nats connection method which you can find here: https://docs.tacticalrmm.com/functions/settings_override/#use-nats-standard-instead-of-nats-websocket
